### PR TITLE
Update jobs.json to latest generated swagger file as well as it's updated examples

### PR DESCRIPTION
**Pull request by johnpaulkee**
# Summary

Updated sql readme.md to include jobs.json where necessary (will need confirmation if this is correctly placed)

Regenerated swagger to include job step and target executions, sku name and capacity support in agent createOrUpdate and get, as well as filter for top executions.

Job examples were updated to reflect correct json request and response paramters for agent and target group.

# Review checklist

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
- [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

( I ran this: oav validate-spec jobs.json --azure-validator and it gave no errors were found )

### CODE REVIEW

**Comment by hovsepm** [comment]
```
  "id": "1026",
  "code": "TypeChanged",
  "message": "The new version has a different type 'integer' than the previous one 'string'.",
  "jsonref": "#/paths/~1subscriptions~1{subscriptionId}~1resourceGroups~1{resourceGroupName}~1providers~1Microsoft.Sql~1servers~1{serverName}~1jobAgents~1{jobAgentName}~1executions/get/$skip",
  "json-path": "#/paths/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/jobAgents/{jobAgentName}/executions/get/$skip",
  "type": "Error"
```

You are introducing breaking changes to all the generated client SDKs. Should this be combined with API version bump PR?

**Comment by johnpaulkee** [comment]
@hovsepm I think it's fine to keep this breaking change, but I think before we merge this PR I'll wait for @jaredmoo to sign off on this too since he's almost back from vacation.

**Comment by jaredmoo** [comment]
This API is not yet published in the SDK, so there is technically no breaking change here.

File: specification/sql/resource-manager/readme.md
Comment (jaredmoo) [inline specification/sql/resource-manager/readme.md]: Please don't add to readme.md until we are ready to publish SDK.

**Review by jaredmoo — CHANGES_REQUESTED** [review]
Looks great, just undo readme.md plz

**Comment by johnpaulkee** [comment]
Undid readme.md, updating title to remove do not merge.

**Review by jaredmoo — APPROVED** [review]
Thanks!

**Comment by jaredmoo** [comment]
It has been reviewed internally by email and already lit up, but I don't think @ravbhatnagar has explicitly approved on github.

**Comment by hovsepm** [comment]
@ravbhatnagar could you take a look to the entire swagger? We never got a sign-off from ARM.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/jobs.json
Comment (ravbhatnagar) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/jobs.json]: GET on /targets or /steps returns the same entity - JobExecution?

**Comment by ravbhatnagar** [comment]
Signing off!

**Comment by hovsepm** [comment]
@johnpaulkee @jaredmoo Please add the readme.md changes back and keep this PR open until your service will be deployed and SDK's could be generated.

**Comment by hovsepm** [comment]
Also please fix linter error - https://travis-ci.org/Azure/azure-rest-api-specs/jobs/369280805

**Comment by hovsepm** [comment]
Please fix sample issues reported in sample validation - https://travis-ci.org/Azure/azure-rest-api-specs/jobs/369280815

**Comment by jaredmoo** [comment]
The linter error is expected. Operation API is defined in a different file (operations.json)

The `Missing required property: password` model errors are all bugs in validator. "Password" is a required write-only property and it is intentionally null in responses.

The tablename errors (e.g. https://travis-ci.org/Azure/azure-rest-api-specs/jobs/369280815#L900 ) might be legit. @johnpaulkee I'll let you check it out.

**Comment by jaredmoo** [comment]
Filed https://github.com/Azure/oav/issues/239 for the OAV false positive.

**Comment by johnpaulkee** [comment]
I've also removed [Do not merge] from the title as we're intending on checking this in since we're getting close to public preview and we've received ARM sign off.

**Review by jaredmoo — APPROVED** [review]
Minor comments. Thanks!

File: specification/sql/resource-manager/readme.md
Comment (jaredmoo) [reply specification/sql/resource-manager/readme.md]: In other words: please delete the new line 54 & 55 :)

Comment (johnpaulkee) [reply specification/sql/resource-manager/readme.md]: You got it :)

File: specification/sql/resource-manager/readme.md
Comment (johnpaulkee) [reply specification/sql/resource-manager/readme.md]: Will update to be in alphabetical order.

**Comment by johnpaulkee** [comment]
Updated based on feedback from @jaredmoo.
@hovsepm when you get a chance can you review once more?
We're ready to publish this :)

**Review by hovsepm — APPROVED** [review]

**Pull request merged**
